packages/backend/src/workers/provider/InferenceProvider.ts

```diff
@@ -0,0 +1,105 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type {
+  ContainerCreateOptions,
+  ContainerCreateResult,
+  Disposable,
+  ImageInfo,
+  PullEvent,
+} from '@podman-desktop/api';
+import { containerEngine } from '@podman-desktop/api';
+import type { InferenceServerConfig } from '@shared/src/models/InferenceServerConfig';
+import type { IWorker } from '../IWorker';
+import type { TaskRegistry } from '../../registries/TaskRegistry';
+import { getImageInfo, getProviderContainerConnection } from '../../utils/inferenceUtils';
+
+export type BetterContainerCreateResult = ContainerCreateResult & { engineId: string };
+
+export abstract class InferenceProvider
+  implements IWorker<InferenceServerConfig, BetterContainerCreateResult>, Disposable
+{
+  protected constructor(private taskRegistry: TaskRegistry) {}
+
+  abstract name: string;
+  abstract enabled(): boolean;
```

Comment:
Not sure what is the purpose of this ? May need to be async for future Kubernetes provider

Reply:
Because we are extending the IWorker interface, which aims to provide a quick way of determining if a worker is enabled or not.

In the case of a kubernetes connection, the worker is not responsible of maintaining the connection with the cluster, we would have to have a component/registry keeping track of this information, and making it available in a sync maner.